### ISSUE TITLE
question about the python tool 

### ISSUE THREAD

**Issue by zhji2822** [wyfsh/YOLO-LISA#8]
I have some questions about how to use the python tool to get the dataset format that we can use. In the lisa.names file we find there were 5 labels that we need:speedLimit, warning, noTurn, stop, pedestrianCrossing, signalAhead. I assume we need to use the extractAnnotations.py to get those annotations for the training and here is what I did  
`:~/Desktop/CUelectrical car/DATASET/tools$ python extractAnnotations.py -c [speedLimit, noTurn, stop, pedestrainCrossing, signalAhead]

usage: extractAnnotations.py [-h] [-f acceptedTag] [-c category] [-m 5]
                             {copy,mark,blackout,crop} filename.csv

extractAnnotations.py: error: argument mode: invalid choice: 'noTurn,' (choose from 'copy', 'mark', 'blackout', 'crop')
`
Can you help me to find the way that I can convert the dataset into the format that I need? thank you!!

**Comment by zhji2822** [comment]
This is what I am using right now to run the "python tool" file how I just got the images but I didn't get the annotations that I want. Any help will be appreciated. thank you!!

`python extractAnnotations.py -c warning  copy allAnnotations.csv
`

**Comment by wyfsh** [comment]
@zhji2822 I think you are on the right track. However, this is quite an old project I did almost two years ago and now I am busy with other jobs. You can modify the annotationTools (I remember just a little) downloaded from [LISA website](http://cvrr.ucsd.edu/vivachallenge/index.php/signs/sign-detection/). It seems you have already got that code.

**Comment by wyfsh** [comment]
@zhji2822 I got your problem. The 6 labels (speedLimit, warning, noTurn, stop, pedestrianCrossing, signalAhead) are the superclasses of signs. They are defined in the [LISA](http://cvrr.ucsd.edu/vivachallenge/index.php/signs/sign-detection/) project papers. You can google their lab's papers.
Because of the limited computing power, we did this simplication to detect faster and more accurate. Of course, you can try to deal with the more general one.

**Comment by zhji2822** [comment]
yeah, I think you are right. But I still could not find out the way to
create the annotated file with Yolo format. You mentioned that we need to
edit the python by ourselves. And I am confused how to do that yet.

On Dec 27, 2018 9:42 AM, "Yifan Wang" <notifications@github.com> wrote:

@zhji2822 <https://github.com/zhji2822> I got your problem. The 6 labels
(speedLimit, warning, noTurn, stop, pedestrianCrossing, signalAhead) are
the superclasses of signs. They are defined in the LISA
<http://cvrr.ucsd.edu/vivachallenge/index.php/signs/sign-detection/>
project papers. You can google their lab's papers.
Because of the limited computing power, we did this simplication to detect
faster and more accurate. Of course, you can try to deal with the general
one.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/wyfsh/YOLO-LISA/issues/8#issuecomment-450187644>, or mute
the thread

**Comment by zhji2822** [comment]
I tried crop, copy, Mark, and blackout all last night. And they only output
images.

**Comment by wyfsh** [comment]
@zhji2822 You should carefully read the first 10 lines of `evaluateDetections.py` and `filterAnnotationFile.py`.
`[-c category]`  -- `category` should be a txt file.

**Comment by zhji2822** [comment]
Thank you!!
